Olaheavy2021/registrationApp
# CBP-32: Implement Forgot Password Functionality for the Users

# Ticket Number and description
[Implement Forgot Password Functionality for the Users](https://sunkanmiolawuwogas.atlassian.net/browse/CBP-32)

## Context
As a user, I want the ability to reset my password in case I forget it or need to change it for security reasons. This will allow me to regain access to my account and ensure the security of my personal information. 

## Work Done

- Provide a form where users can enter their registered email address to request a password reset.
- Validate the email address to ensure it belongs to an existing user account.
- Generate a unique password reset token and associate it with the user account.
- Send an email to the user's registered email address containing a link with the password reset token.
- Create a "Reset Password" page where users can enter a new password after clicking the password reset link in the email.
- Implement the logic to update the user's password in the database once the new password is submitted.

## Screenshots

Insert any relevant screenshots with captions for context

## Testing instructions

- Navigate to the Login Page
- Click on Forgot Password
- Enter Your Email Associated to your Student Account
- You should receive an email
- Click on the Link Provided in the email
- Enter the new Password
- Navigate to the Login Page and try out the new password

## Acceptance Criteria

- [x] Users can initiate the password reset process by providing their registered email address.
- [x] A password reset email with a unique token is sent to the user's registered email address.
- [x] The "Reset Password" page allows users to enter and confirm a new password.
- [x] The new password is securely updated in the database after successful submission.
